shared/constants/abi.ts

```diff
@@ -0,0 +1,181 @@
+export const proposePriceAbi = [
```

Comment:
these abis are copied from etherscan. I think this is okay because I don't expect these to change, but we can make another plan if that's not gonna work